### PR TITLE
MM-26064 - ENV var license override

### PR DESCRIPTION
**Pull request by wiggin77**
#### Summary

This PR adds support for a new environment variable `MM_LICENSE` which can contain the contents of a license file.  When set, this license takes priority over all other license sources.  

- Database and file licenses are not overwritten but simply ignored while `MM_LICENSE` is set.

- If MM_LICENSE is set to a non-empty string, but the license is not valid, the server will be started without a license.

#### Ticket Link

https://mattermost.atlassian.net/browse/MM-26064

### CODE REVIEW

File: app/license.go [hunk 1]
```diff
@@ -7,16 +7,29 @@ import (
...
+	requestTrialURL = "https://customers.mattermost.com/api/v1/trials"
+	LicenseENV      = "MM_LICENSE"
+)
 
 func (s *Server) LoadLicense() {
+	// ENV var overrides all other sources of license.
+	licenseStr := os.Getenv(LicenseENV)
+	if licenseStr != "" {
+		if s.ValidateAndSetLicenseBytes([]byte(licenseStr)) {
+			mlog.Info("License key from ENV valid unlocking enterprise features.")
```
Comment (reflog) [inline app/license.go:28]: small nit: 
```suggestion
			mlog.Info("License key from ENV is valid, unlocking enterprise features.")
```

File: app/license.go [hunk 2]
```diff
@@ -7,16 +7,29 @@ import (
...
 
 	"github.com/mattermost/mattermost-server/v5/mlog"
 	"github.com/mattermost/mattermost-server/v5/model"
 	"github.com/mattermost/mattermost-server/v5/utils"
 )
 
-const requestTrialURL = "https://customers.mattermost.com/api/v1/trials"
+const (
+	requestTrialURL = "https://customers.mattermost.com/api/v1/trials"
+	LicenseENV      = "MM_LICENSE"
```
Comment (jwilander) [inline app/license.go:20]: nit: should this be `LicenseEnv`? 0/5

**Comment by jwilander** [comment]
cc @cpanato

File: app/license.go (hunk 2 above)
Comment (cpanato) [reply app/license.go:20]: What happens with the current settings we have? for example, if the user sets the `ServiceSettings.LicenseFileLocation` and this new var?

side note: in the operator side, we will need to manage two ways to inject the license for versions `<= 5.25`

Comment (wiggin77) [reply app/license.go:20]: The ENV var takes priority over LicenseFileLocation and any DB saved license. In other words, those are ignored as long as MM_LICENSE is defined with a non-empty string.

**Comment by wiggin77** [comment]
@lindy, is this something that needs QA?

**Comment by amyblais** [comment]
@wiggin77 I noticed this wasn't cherry-picked. Is this too risky to include in v5.25 and would be better to push to v5.26?

**Comment by wiggin77** [comment]
> @wiggin77 I noticed this wasn't cherry-picked. Is this too risky to include in v5.25 and would be better to push to v5.26?

This is a new feature, therefore I didn't think was a candidate for a stability release.

**Comment by amyblais** [comment]
@wiersgallak I've pushed this to v5.26 since this is a new feature and we're past code complete and RC testing. Let me know if you'd like to re-discuss adding this to v5.25.

**Comment by wiersgallak** [comment]
@amyblais thanks for the heads up.  5.26 is fine.

**Comment by cwarnermm** [comment]
Documentation updated via: https://github.com/mattermost/docs/pull/4204

**Pull request merged**